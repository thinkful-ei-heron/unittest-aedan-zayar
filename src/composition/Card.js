import React from 'react';

class Card extends React.Component {
    render() {
        return (<div className="Card">
                    <button className="button">
                        delete
                    </button>
                    <h3>
                        {this.props.title}
                    </h3>
                    <p>
                        {this.props.message}
                    </p>
                </div>)
    }
}

export default Card;