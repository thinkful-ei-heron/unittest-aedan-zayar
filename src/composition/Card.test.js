import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card.js';

describe('Card component', () => {
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Card title="Test title" message="Test message" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Card title="Test Title" message="Test message" />, div);
      ReactDOM.unmountComponentAtNode(div);
    })
});