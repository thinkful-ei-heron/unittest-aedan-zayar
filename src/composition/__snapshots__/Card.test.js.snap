exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card">
  <button
    className="button">
    delete
  </button>
  <h3>
    Test title
  </h3>
  <p>
    Test message
  </p>
</div>
`;

exports[`List component renders the UI as expected 1`] = `
<div
  className="Card">
  <button
    className="button">
    delete
  </button>
  <h3>
    Test title
  </h3>
  <p>
    Test message
  </p>
</div>
`;
